core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -69,7 +69,7 @@
     this.schema = spec.schema();
     this.nameToField = indexSpecByName(spec);
     this.transformToField = indexSpecByTransform(spec);
-    this.lastAssignedPartitionId = base.lastAssignedPartitionId();
+    this.lastAssignedPartitionId = base.lastPartitionId();
```

Comment:
I don't think we need to rename the method, just the property in the metadata file.